### PR TITLE
Added facingMode parameter to choose camera type

### PR DESCRIPTION
**Pull request by agustencp**
To choose the type of camera, especially on mobile devices (selfie or environment camera) without knowing the deviceId.

### CODE REVIEW

File: package/lib/index.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ export interface QrCodeReaderProps {
   onRead?: (code: QRCode) => void;
   action?: (str: string) => void;
   deviceId?: string;
+  facingMode?: string;
```
Comment (ysuzuki19) [inline package/lib/index.tsx]: Could you change `videoConstraints`?

And remove line of `deviceId`.

File: package/lib/index.tsx [hunk 2]
```diff
@@ -56,7 +58,7 @@ function QrCodeReader({
       ref={webcamRef}
       width={width}
       height={height}
-      videoConstraints={{ deviceId }}
+      videoConstraints={{ deviceId, facingMode }}
```
Comment (ysuzuki19) [inline package/lib/index.tsx]: With the change above,

Please change to input `videoConstraints`.

**Review by ysuzuki19 — CHANGES_REQUESTED** [review]
Hi, @agustencp ,
Thank you for your PR!

I would like to deprecate the `deviceId` prop and integrate it into `videoConstraints`.

**Comment by ysuzuki19** [comment]
And I would like to update doc(update `README`, add `MIGRATION.md`).

Shall I do it?

File: package/lib/index.tsx (hunk 1 above)
Comment (agustencp) [reply package/lib/index.tsx]: ok, I will update it

**Comment by agustencp** [comment]
> And I would like to update doc(update `README`, add `MIGRATION.md`).
> 
> Shall I do it?

Right, `README` is ignored, that's why it didn't get on this PR

**Comment by ysuzuki19** [comment]
Thank you the change
I'll see the code tonight (from Tokyo)

**Comment by ysuzuki19** [comment]
Oh sorry, i didn't explain enough.

This repository has two meaning of `README`, that are "README for GitHub" and "README for npm".

In `lib/package.json`, the script `npm run prepublish` copy some files at root of repository for npm package.

So you can update `README` of npm package by updating `README` at root of repository without updating `.gitignore`.

**Comment by agustencp** [comment]
> Oh sorry, i didn't explain enough.
> 
> This repository has two meaning of `README`, that are "README for GitHub" and "README for npm".
> 
> In `lib/package.json`, the script `npm run prepublish` copy some files at root of repository for npm package.
> 
> So you can update `README` of npm package by updating `README` at root of repository without updating `.gitignore`.

ok, done

**Comment by ysuzuki19** [comment]
okay, the code looks like good!

Can you restore `lib/package-lock.json`?

After that, I'll verify operation on my devices and merge it.

**Comment by agustencp** [comment]
> okay, the code looks like good!
> 
> Can you restore `lib/package-lock.json`?
> 
> After that, I'll verify operation on my devices and merge it.

Done!

**Review by ysuzuki19 — APPROVED** [review]

**Comment by ysuzuki19** [comment]
@agustencp 
Thanks changing.

The code looks good and I verified operation on chrome and ios-safari (Simulator).

**Comment by ysuzuki19** [comment]
plan to do, I will merge it and publish new version at this weekend.

**Pull request merged**